### ISSUE TITLE
Doesn't catch CodePipeline OutputArtifacts need to be uniquely named

### ISSUE THREAD

**Issue by Graham42** [aws-cloudformation/cfn-python-lint#1688]
cfn-lint 0.35.1

*Description of issue.*
The linter doesn't catch that CodePipeline `OutputArtifacts` need to be uniquely named.

Please provide as much information as possible:
* Template linting issues: 
  * Please provide a CloudFormation sample that generated the issue.

This template generates the error `UPDATE_FAILED | Output Artifact Bundle name must be unique within the pipeline. CreateOutput has been used more than once.`

<details>

```yaml
AWSTemplateFormatVersion: "2010-09-09"
Description: The AWS CloudFormation template for this Serverless application
Resources:
  ServerlessDeploymentPipeline:
    Type: AWS::CodePipeline::Pipeline
    Properties:
      ArtifactStores:
        - Region: ca-central-1
          ArtifactStore:
            Type: S3
            Location: my-artifact-bucket
      Name: my-code-pipeline
      RestartExecutionOnUpdate: false
      RoleArn: arn:aws:iam::000000000000:role/root
      Stages:
        - Name: Source
          Actions:
            - Name: SourceAction
              ActionTypeId:
                Category: Source
                Owner: AWS
                Version: "1"
                Provider: S3
              OutputArtifacts:
                - Name: SourceArtifact
              Configuration:
                S3Bucket: my-source-bucket
                S3ObjectKey: source-item.zip
              RunOrder: 1
        - Name: DeployToEnvA
          Actions:
            - Name: CreateChangeSetEnvA
              Region: us-east-1
              ActionTypeId:
                Category: Deploy
                Owner: AWS
                Version: "1"
                Provider: CloudFormation
              InputArtifacts:
                - Name: SourceArtifact
              OutputArtifacts:
                - Name: CreateOutput
              Configuration:
                ActionMode: CHANGE_SET_REPLACE
                StackName: my-service-env-a
                Capabilities: CAPABILITY_NAMED_IAM
                RoleArn: arn:aws:iam::000000000000:role/root
                TemplatePath: SourceArtifact::env-a-us-east-1.json
                ChangeSetName: ChangeSet
              RunOrder: 1
              RoleArn: arn:aws:iam::000000000000:role/root
            - Name: CreateChangeSetEnvB
              Region: us-east-1
              ActionTypeId:
                Category: Deploy
                Owner: AWS
                Version: "1"
                Provider: CloudFormation
              InputArtifacts:
                - Name: SourceArtifact
              OutputArtifacts:
                - Name: CreateOutput
              Configuration:
                ActionMode: CHANGE_SET_REPLACE
                StackName: my-service-env-b
                Capabilities: CAPABILITY_NAMED_IAM
                RoleArn: arn:aws:iam::000000000000:role/root
                TemplatePath: SourceArtifact::env-b-us-east-1.json
                ChangeSetName: ChangeSet
              RunOrder: 1
              RoleArn: arn:aws:iam::000000000000:role/root

```

</details>

  * If present, please add links to the (official) documentation for clarification.
      - > Every output artifact in the pipeline must have a unique name.  

          [Source](https://docs.aws.amazon.com/codepipeline/latest/userguide/welcome-introducing-artifacts.html)
  * Validate if the issue still exists with the latest version of `cfn-lint` and/or the latest Spec files: :heavy_check_mark:  `0.35.1` is the latest version

Cfn-lint uses the [CloudFormation Resource Specifications](https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/cfn-resource-specification.html) as the base to do validation. These files are included as part of the application version. Please update to the latest version of `cfn-lint` or update the spec files manually (`cfn-lint -u`)

:heavy_check_mark:  I have also tried after running `cfn-lint -u`
